[CI] StatelessRealTimeGetIT testStress failing

**Build Scans:**
- [elasticsearch-intake #40784 / part2](https://gradle-enterprise.elastic.co/s/b46gwkbmdeqoq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT.testStress" -Dtests.seed=AEBCBF8C0CABABD -Dtests.locale=as-Beng-IN -Dtests.timezone=Asia/Kashgar -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testStress'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Shard [test-index][0] is still locked after 5 sec waiting
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testStress
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -309,9 +309,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.engine.RefreshNodeCreditManagerTests
   method: testPeriodicWarning
   issue: https://github.com/elastic/elasticsearch-serverless/issues/6518
-- class: org.elasticsearch.xpack.stateless.StatelessRealTimeGetIT
-  method: testStress
-  issue: https://github.com/elastic/elasticsearch/issues/147850
 - class: org.elasticsearch.backwards.RareTermsIT
   method: testSingleValuedString
   issue: https://github.com/elastic/elasticsearch/issues/147853

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/ConcurrentMultiPartUploadsMockFsRepository.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/ConcurrentMultiPartUploadsMockFsRepository.java
@@ -224,7 +224,6 @@ class ConcurrentMultiPartUploadsMockFsRepository extends FsRepository {
                                             format("[%d] bytes copied for writing part, expecting [%d] bytes", bytesCopied, partSize)
                                         );
                                     }
-
                                 }));
                                 remaining -= partSize;
                                 assert 0 <= remaining : remaining;
@@ -244,7 +243,9 @@ class ConcurrentMultiPartUploadsMockFsRepository extends FsRepository {
                             if (assertion.getCause() instanceof ExecutionException ee && ee.getCause() instanceof IOException ioe) {
                                 throw ioe;
                             }
-                            throw assertion;
+                            // For all other cases, consumers of writeBlobAtomic expect IOExceptions,
+                            // not AssertionErrors, so convert here to meet that contract.
+                            throw new IOException("multipart upload failed", assertion);
                         }
 
                         if (bytesWritten.get() != blobSize) {

